packit/config.py

```diff
@@ -113,12 +115,15 @@ def is_dict_valid(cls, raw_dict: dict) -> bool:
         return Draft4Validator(JOB_CONFIG_SCHEMA).is_valid(raw_dict)
 
 
-@dataclass(unsafe_hash=True, frozen=True)
 class PackageConfig:
-    specfile_path: str
-    synced_files: List[str]
-    jobs: List[JobConfig]
-    metadata: dict
+    def __init__(self):
+        self.specfile_path: Optional[str] = None
+        self.synced_files: Optional[List[str]] = None
+        self.jobs: Optional[List[JobConfig]] = None
+        # TODO: the metadata should have a proper definition and validation
```

Comment:
my point here is: this felt really wrong:
```
self.config.package_config.metadata["something"]
```
I don't wanna access a dict when we have a full-blown config module.

Reply:
Yes, I only prefer extracting the known parameters from `metadata` and avoid structuring the `metadata` attribute. (I just want to leave the `metadata` for the unknown info in the config file.)

Reply:
we shouldn't have unknown info in the config file

Reply:
E.g. upstream url:
It can be found with upstream-release-monitoring, but for now, we have it in the config file.

From the long-term view, I agree that unknown values are not needed.